nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/QueryDatabaseTable.java

```diff
@@ -437,7 +439,7 @@ public void processRow(ResultSet resultSet) throws IOException {
                 if (nrOfColumns > 0) {
                     for (int i = 1; i <= nrOfColumns; i++) {
                         String colName = meta.getColumnName(i).toLowerCase();
-                        String fullyQualifiedMaxValueKey = getStateKey(meta.getTableName(i), colName);
+                        String fullyQualifiedMaxValueKey = getStateKey(tableName, colName);
```

Comment:
I found a discussion on this from way back in 2007 where it sounded like it depended on which cursor type was used to bring back the results, and which column types were included.
https://social.msdn.microsoft.com/Forums/sqlserver/en-US/55e8cbb2-b11c-446e-93ab-dc30658caf99/resultsetmetadatagettablename-returns-instead-of-table-name?forum=sqldataaccess